esp-mbedtls/src/lib.rs

```diff
@@ -495,30 +535,161 @@ impl<'a> Certificates<'a> {
     }
 }
 
-pub struct Session<T> {
+/// A TLS self-test type
+#[derive(enumset::EnumSetType, Debug)]
+pub enum TlsTest {
+    Mpi,
+    Rsa,
+    Sha1,
+    Sha224,
+    Sha256,
+    Sha384,
+    Sha512,
+    Aes,
+    Md5,
+}
+
+impl fmt::Display for TlsTest {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            TlsTest::Mpi => write!(f, "MPI"),
+            TlsTest::Rsa => write!(f, "RSA"),
+            TlsTest::Sha1 => write!(f, "SHA1"),
+            TlsTest::Sha224 => write!(f, "SHA224"),
+            TlsTest::Sha256 => write!(f, "SHA256"),
+            TlsTest::Sha384 => write!(f, "SHA384"),
+            TlsTest::Sha512 => write!(f, "SHA512"),
+            TlsTest::Aes => write!(f, "AES"),
+            TlsTest::Md5 => write!(f, "MD5"),
+        }
+    }
+}
+
+static TLS_CREATED: Mutex<Cell<bool>> = Mutex::new(Cell::new(false));
+
+/// A TLS instance
+///
+/// Represents an instance of the MbedTLS library.
+/// Only one such instance can be active at any point in time.
+pub struct Tls<'d>(PhantomData<&'d mut ()>);
+
+impl<'d> Tls<'d> {
+    /// Create a new instance of the `Tls` type.
+    ///
+    /// Note that there could be only one active `Tls` instance at any point in time,
+    /// and the function will return an error if there is already an active instance.
+    #[cfg(all(not(any(
+        feature = "esp32",
+        feature = "esp32c3",
+        feature = "esp32s2",
+        feature = "esp32s3"
+    ))))]
+    pub fn new() -> Result<Self, TlsError> {
+        Self::create()
+    }
+
+    pub(crate) fn create() -> Result<Self, TlsError> {
+        critical_section::with(|cs| {
+            let created = TLS_CREATED.borrow(cs).get();
+
+            if created {
+                return Err(TlsError::AlreadyCreated);
+            }
+
+            TLS_CREATED.borrow(cs).set(true);
+
+            Ok(Self(PhantomData))
+        })
+    }
+
+    /// Set the MbedTLS debug level (0 - 5)
+    #[allow(unused)]
+    pub fn set_debug(&mut self, level: u32) {
+        #[cfg(all(not(target_os = "espidf"), not(target_arch = "xtensa")))]
```

Comment:
I think this check can be removed, since the underlying issue with C variadics has been resolved for Xtensa

Reply:
Done: I've removed the check (only left it in-place for `espidf` as that version of the lib is compiled without debug support).

I also enabled the "DEBUG_C" macro in all xtensa configs and re-ran the `xtask`s for all targets.